ocaml/xapi/xapi_cluster_host.ml

```diff
@@ -112,17 +112,25 @@ let join_internal ~__context ~self =
       let cluster_token =
         Db.Cluster.get_cluster_token ~__context ~self:cluster
       in
-      let ip = ip_of_pif (pIF, Db.PIF.get_record ~__context ~self:pIF) in
+      let ip_addr = ip_of_pif (pIF, Db.PIF.get_record ~__context ~self:pIF) in
+      let hostid = Inventory.lookup Inventory._installation_uuid in
```

Comment:
I assume this always runs locally on the pool member?

Reply:
you mean the inventory lookup? I think so because it seems that the lookup is just a hashtable lookup of a local file...

Reply:
https://github.com/xapi-project/xcp-inventory/blob/26e78d63c9462d86bc79865d91dc8e6c0ef96f99/lib/inventory.ml#L95-L104

Reply:
The inventory lookup is always local yes. I just wanted to make sure that this function is always running on the host that we need to get the UUID from. I assume so, as the coordinator has passed the join call to the host that is joining at this point? So other info comes from the DB, so it doesn't matter where in the pool you run it, but this inventory lookup has to be done on the right host.

Reply:
oh I see. I would think so from the result of the manual test. The code below under `pool_create` should call functions on each host? And each of them, in `create`, would then call `join_internal`

```ocaml
  try
    List.iter
      (fun host ->
        (* Cluster.create already created cluster_host on master, so we only iterate through slaves *)
        Helpers.call_api_functions ~__context (fun rpc session_id ->
            let pif, _ = pif_of_host ~__context network host in
            let cluster_host_ref =
              Client.Client.Cluster_host.create ~rpc ~session_id ~cluster ~host
                ~pif
            in
            D.debug "Created Cluster_host: %s" (Ref.string_of cluster_host_ref)
        )
      )
      slave_hosts ;
    cluster
 ```